### PR TITLE
Automatically transmit some headers specified by a environment varaible that directly defines header keys.

### PR DESCRIPTION
**Pull request by pandaapo**
## PR Type
Feature.

## Describe what this PR does for and how you did.
Automatically transmit some headers specified by a environment varaible that directly defines header keys.
Use the `MetadataContext` to build headers needed to be transmitted. When current service sends request, attach the transitive headers saved in `MetadataContext` before.

## Adding the issue link (#xxx) if possible.
#563

## Note
Junit tests will be made up later.

## Checklist

- [ ] Add copyright holder at the beginning of .class file if it is new.
- [x] Add information of this PR to CHANGELOG.md in root of project.
- [ ] All junit tests passing.
- [ ] Coverage from `Codecov Report` should not decrease.

## Checklist (Optional)

- [x] Will Pull Request to branch of hoxton.
- [x] Add documentation in javadoc in code or comment below the PR if necessary.

### CODE REVIEW

**Comment by andrewshan** [comment]
测试用例不通过，麻烦确认下

**Comment by leopardrota** [comment]
好。没来得及写，还是旧的测试代码。

File: spring-cloud-starter-tencent-metadata-transfer/src/main/java/com/tencent/cloud/metadata/core/DecodeTransferMetadataReactiveFilter.java [hunk 1]
```diff
@@ -79,12 +87,41 @@ public Mono<Void> filter(ServerWebExchange serverWebExchange, WebFilterChain web
...
+	 * According to ServerHttpRequest and trans-headers(key list in string type) in metadata, build
+	 * the complete headers(key-value list in map type) into metadata.
+	 */
+	private void setCompleteTransHeaderIntoMC(ServerHttpRequest serverHttpRequest) {
+		// transHeaderMetadata: for example, {"trans-headers" : {"header1;header2;header3":""}}
+		Map<String, String> transHeaderMetadata =  MetadataContextHolder.get()
+				.getFragmentContext(FRAGMENT_RAW_TRANSHEADERS);
+		if (!CollectionUtils.isEmpty(transHeaderMetadata)) {
+			Optional<String> transHeaders = transHeaderMetadata.keySet().stream().findFirst();
+			String[] transHeaderArray = transHeaders.get().split(";");
```
Comment (chuntaojun) [inline spring-cloud-starter-tencent-metadata-transfer/src/main/java/com/tencent/cloud/metadata/core/DecodeTransferMetadataReactiveFilter.java:107]: Optional.get() 会抛 Null 异常，使用 orElse 避免异常吧

File: spring-cloud-starter-tencent-metadata-transfer/src/main/java/com/tencent/cloud/metadata/core/DecodeTransferMetadataServletFilter.java [hunk 2]
```diff
@@ -71,9 +77,35 @@ protected void doFilterInternal(@NonNull HttpServletRequest httpServletRequest,
...
+	 * According to HttpServletRequest and trans-headers(key list in string type) in metadata, build
+	 * the complete headers(key-value list in map type) into metadata.
+	 */
+	private void setCompleteTransHeaderIntoMC(HttpServletRequest httpServletRequest) {
+		// transHeaderMetadata: for example, {"trans-headers" : {"header1;header2;header3":""}}
+		Map<String, String> transHeaderMetadata =  MetadataContextHolder.get()
+				.getFragmentContext(FRAGMENT_RAW_TRANSHEADERS);
+		if (!CollectionUtils.isEmpty(transHeaderMetadata)) {
+			Optional<String> transHeaders = transHeaderMetadata.keySet().stream().findFirst();
+			String[] transHeaderArray = transHeaders.get().split(";");
```
Comment (chuntaojun) [inline spring-cloud-starter-tencent-metadata-transfer/src/main/java/com/tencent/cloud/metadata/core/DecodeTransferMetadataServletFilter.java:94]: 同上

Comment (chuntaojun) [inline spring-cloud-starter-tencent-metadata-transfer/src/main/java/com/tencent/cloud/metadata/core/DecodeTransferMetadataServletFilter.java:94]: 下面的方法抽象到 transfer 吧

**Review by chuntaojun — CHANGES_REQUESTED** [review]

**Comment by pandaapo** [comment]
> 测试用例不通过，麻烦确认下

好。没来得及写，还是之前的测试代码。

File: spring-cloud-tencent-commons/src/main/java/com/tencent/cloud/common/metadata/StaticMetadataManager.java [hunk 3]
```diff
@@ -55,6 +55,12 @@ public class StaticMetadataManager {
 	private static final int ENV_METADATA_PREFIX_LENGTH = ENV_METADATA_PREFIX.length();
 	private static final String ENV_METADATA_CONTENT_TRANSITIVE = "SCT_METADATA_CONTENT_TRANSITIVE";
 	private static final String ENV_METADATA_CONTENT_DISPOSABLE = "SCT_METADATA_CONTENT_DISPOSABLE";
+	/**
+	 * This is the key of the header's key list needed to be transmitted. The list is a string split with ;.
+	 * The value mapped by this key was specified by user.
+	 * This is configured in environment variables.
+	 */
+	private static final String ENV_METADATA_CONTENT_RAW_TRANSHEADERS = "SCT_METADATA_CONTENT_RAW_TRANSHEADERS";
```
Comment (chuntaojun) [inline spring-cloud-tencent-commons/src/main/java/com/tencent/cloud/common/metadata/StaticMetadataManager.java:63]: 这里新增的这个信息，会注册到北极星的实例中去，这个可能需要处理下，这个是不需要注册到实例的元数据中的

**Review by chuntaojun — CHANGES_REQUESTED** [review]

**Comment by andrewshan** [comment]
> 的

会导致之前的测试代码跑不过吗？

File: spring-cloud-starter-tencent-metadata-transfer/src/main/java/com/tencent/cloud/metadata/core/DecodeTransferMetadataReactiveFilter.java (hunk 1 above)
Comment (pandaapo) [reply spring-cloud-starter-tencent-metadata-transfer/src/main/java/com/tencent/cloud/metadata/core/DecodeTransferMetadataReactiveFilter.java:107]: OK

File: spring-cloud-starter-tencent-metadata-transfer/src/main/java/com/tencent/cloud/metadata/core/DecodeTransferMetadataServletFilter.java (hunk 2 above)
Comment (pandaapo) [reply spring-cloud-starter-tencent-metadata-transfer/src/main/java/com/tencent/cloud/metadata/core/DecodeTransferMetadataServletFilter.java:94]: OK

Comment (pandaapo) [reply spring-cloud-starter-tencent-metadata-transfer/src/main/java/com/tencent/cloud/metadata/core/DecodeTransferMetadataServletFilter.java:94]: 不太确定您的意思，是要将这段在多个类之间重复的代码提取成工具类公用代码吗？

File: spring-cloud-tencent-commons/src/main/java/com/tencent/cloud/common/metadata/StaticMetadataManager.java (hunk 3 above)
Comment (pandaapo) [reply spring-cloud-tencent-commons/src/main/java/com/tencent/cloud/common/metadata/StaticMetadataManager.java:63]: 不好意思，没来及处理。感谢帮助处理了。

**Pull request merged**